site/public/js/server.js

```diff
@@ -133,33 +141,79 @@ function changeDiffView(div_name, gradeable_id, who_id, version, index, autochec
 
 }
 
+
+/**
+ * Collapses Test Case Output.  
+ *
+ * @param {string} div_name - id of div containing the output of the test case
+ * @param {string} index - index of test case
+ * @param {object} loadingTools - loading tools used by span containing the text that 
+ *                                correspond to the state of the test case (show, hide, loading)
+ */
+function CollapseTestcaseOutput(div_name, index, loadingTools)
+{
+    $("#show_char_" + index).toggle();
+    $("#" + div_name).empty();
+    toggleDiv(div_name);
+
+    loadingTools.find("span").hide();
+    loadingTools.find(".loading-tools-show").show();
+}
+
+/**
+ * Expands or Collapses Test Case Output.  
+ * 
+ * @global {output_xhrs} : Map (Key - string [orig_div_name]; Value - XMLHttpRequest object) to store the
+ *                         XMLHttpRequest object returned by getJSON when getting the output to display
+ * 
+ * @param {string} div_name - id of div containing the output of the test case
+ * @param {string} gradeable_id - ID of the gradeable
+ * @param {string} who_id - ID of the submitter
+ * @param {string} index - index of test case
+ * @param {number} version - submission version
+ */
 function loadTestcaseOutput(div_name, gradeable_id, who_id, index, version = ''){
     let orig_div_name = div_name;
     div_name = "#" + div_name;
 
     let loadingTools = $("#tc_" + index).find(".loading-tools");
 
-    if($(div_name).is(":visible")){
-        $("#show_char_"+index).toggle();
-        $(div_name).empty();
-        toggleDiv(orig_div_name);
-
-        loadingTools.find("span").hide();
-        loadingTools.find(".loading-tools-show").show();
+    // Checks if output for this div is being loaded
+    if(output_xhrs.has(orig_div_name))
+    {
+        // Checks if xhr is defined and has not succeeded yet
+        // If so, abort loading
+        var xhr = output_xhrs.get(orig_div_name);
+        if(xhr && xhr.readyState != 4){
```

Comment:
It is generally better to use the [strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) operators in JS code to prevent unexpected behavior as a result of the default JS type conversion.
```suggestion
        if(xhr && xhr.readyState !== 4){
```